lib/patches/abstract_mysql_adapter.rb

```diff
@@ -0,0 +1,13 @@
+# TODO:
+# MySQL 5.7+ does not allow NULL primary keys, but older versions of Rails
+# create primary key columns DEFAULT NULL.
+#
+# Remove this file and its environment.rb reference after upgrading to Rails 4.
+#
+# https://github.com/rails/rails/pull/13247#issuecomment-32425844
+
+raise "Remove #{__FILE__} for Rails 4+" unless Rails.version < "4"
```

Comment:
`unless Rails.version < "4"` feels a little awkward. Would `if Rails.version >= "4"` be easier to process?